defrag/defrag.go

```diff
@@ -0,0 +1,276 @@
+package defrag
+
+import (
+	"container/list"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"code.google.com/p/gopacket"
+	"code.google.com/p/gopacket/layers"
+)
+
+// Quick and Easy yo use debug code to trace
+// how defrag works.
+var debug debugging = false // or flip to false
+type debugging bool
+
+func (d debugging) Printf(format string, args ...interface{}) {
+	if d {
+		log.Printf(format, args...)
+	}
+}
+
+type Defragmenter interface {
```

Comment:
Well.

1 - I was thinking we would add next a IPv6 defrag, edven if IPv6  is not supposed to be fragment at all. I will rename the package ip4defrag

2 - Regarding the interface, it's the way you specified in the Wishlist ;) Isn't it a good practice to have some high level interface for each major package ? Give me your views on that.